### DISCUSSION TITLE
use RenderSchedule.ScriptProps

### DISCUSSION THREAD

**Discussion by paprophet** [iTwin/itwinjs-core#4196] [Q&A] [answered]
when use RenderSchedule.ScriptProps and move a element from A  position to B  position
but i can not select the key points of the element,for example use the measure tool to select the focal point of the element.

![move](https://user-images.githubusercontent.com/18022720/187585602-cfe023da-e2c6-456a-8f8d-b1092bf8cd71.png)

**Comment by pmconne** [comment]
Snapping is performed on the backend against the persistent geometry. It doesn't know about any arbitrary transforms that may have been applied on the frontend.
We do [account for model display transforms](https://github.com/iTwin/itwinjs-core/blob/4a8ed7a39dddbe3f67526b96ba7989531726c8f6/core/frontend/src/AccuSnap.ts#L720) on the frontend when snapping.
I suppose we ought to be able to do the same for transforms applied to models and/or elements by a schedule script - do you agree, @bbastings?

**Comment by pmconne** [comment, marked as answer]
Tracking in #4198.

**Comment by pmconne** [comment]
@paprophet in your app are you enabling [TileAdmin.Props.enableFrontendScheduleScripts](https://www.itwinjs.org/reference/core-frontend/tiles/tileadmin/tileadmin.props/enablefrontendschedulescripts/)?

**Reply by paprophet** [reply]
yes,i set 
tileAdmin{ enableFrontendScheduleScripts:true,},but  the problem still exists

**Reply by pmconne** [reply]
> but the problem still exists

Right - I just wanted to confirm where your script is coming from. Working on #4198 now.

**Reply by paprophet** [reply]
i set the script data by RenderSchedule.ElementTimelineProps,
in fact, you can set any RenderSchedule.ScriptProps data (must set TransformProps data , also not the Matrix4d.setIdentity().torow()) to tranform element 
this problem include can not  clip the Element by the ViewClipByElementTool in  tranform element  by to set  view.displayStyle.settings.scheduleScriptProps

**Reply by pmconne** [reply]
> can not clip the Element by the ViewClipByElementTool

Yes - that's another case where we ask the backend about the element's geometry (its bounding box) and ignore any transforms applied on the frontend. Thanks for pointing it out. We will address it in #4218.

**Comment by pmconne** [comment]
@paprophet #4198 is complete. It will be available in iTwin.js 3.4, tentatively scheduled for release around the end of September.
If you want to try it out prior to then you can use the 3.4.0-dev.35 packages which will be published within the next 24 hours.

**Comment by paprophet** [comment]
yes , i test and this bug is fix.thank you!
and i have a problem,when i use get the point by PrimitiveTool ,and how i get the transforms value
because i get point from diffent element and have different tranform value,
for example i get first point from A element and A have transfroms value1 and
 get second point from B element and B have transfroms value2,in fact i do not know the point from which element .
how get the point value and transfroms value?

in PrimitiveTool
public override async onDataButtonDown(ev: BeButtonEvent): Promise<EventHandled> 
this method ev have ev.point and transfroms value?

**Reply by pmconne** [reply]
Show your code. Put it inside a code block by like this:
![image](https://user-images.githubusercontent.com/22944042/191000229-6d490217-1806-494a-b961-f83fb3314bbf.png)
